File: pos-scanner-web/backend/models/device_occupancy.py
# backend/models/device_occupancy.py
from extensions import db
from datetime import datetime


class DeviceOccupancy(db.Model):
    """设备占用模型 - 与商家ID关联"""
    __tablename__ = 'device_occupancies'

    id = db.Column(db.Integer, primary_key=True)
    merchant_id = db.Column(db.String(100), unique=True, nullable=False, index=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    purpose = db.Column(db.String(500))  # 用途
    start_time = db.Column(db.DateTime, nullable=False)  # 开始时间
    end_time = db.Column(db.DateTime, nullable=False)  # 结束时间（释放时间）
    created_at = db.Column(db.DateTime, default=datetime.now)

    # 关联用户
    user = db.relationship('User', backref='occupancies')

    def to_dict(self):
        return {
            'merchantId': self.merchant_id,
            'userId': self.user_id,
            'username': self.user.username if self.user else None,
            'purpose': self.purpose,
            'startTime': self.start_time.isoformat() if self.start_time else None,
            'endTime': self.end_time.isoformat() if self.end_time else None,
            'createdAt': self.created_at.isoformat() if self.created_at else None
        }

File: pos-scanner-web/backend-go/internal/models/device_occupancy.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type DeviceOccupancy struct {
	ID         uint           `gorm:"primaryKey" json:"id"`
	MerchantID string         `gorm:"size:100;uniqueIndex;not null" json:"merchant_id"`
	UserID     uint           `gorm:"not null;index" json:"user_id"`
	Purpose    *string        `gorm:"size:500" json:"purpose"`
	StartTime  time.Time      `json:"start_time"`
	EndTime    time.Time      `json:"end_time"`
	CreatedAt  time.Time      `json:"created_at"`
	User       *User          `gorm:"foreignKey:UserID" json:"-"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
}

func (DeviceOccupancy) TableName() string {
	return "device_occupancies"
}

func (d *DeviceOccupancy) ToDict() map[string]interface{} {
	username := ""
	if d.User != nil {
		if d.User.Name != nil && *d.User.Name != "" {
			username = *d.User.Name
		} else {
			username = d.User.Username
		}
	}

	result := map[string]interface{}{
		"merchantId": d.MerchantID,
		"userId":     d.UserID,
		"username":   username,
		"purpose":    d.Purpose,
		"startTime":  d.StartTime.Format(time.RFC3339),
		"endTime":    d.EndTime.Format(time.RFC3339),
		"createdAt":  d.CreatedAt.Format(time.RFC3339),
	}
	return result
}
